### ISSUE TITLE
onnxruntime==1.16.0 release breaks tests

### ISSUE THREAD

**Issue by ksaur** [microsoft/hummingbird#736]
onnxruntime==1.16.0 was [released](https://github.com/microsoft/onnxruntime/releases/tag/v1.16.0)
and [tests](https://github.com/microsoft/hummingbird/actions/runs/6279215738/job/17054486386) fail.

Pinning for now. #737

**Comment by ksaur** [comment]
For the most part, we need to fix:
```
>           raise ValueError(
                f"This ORT build has {available_providers} enabled. "
                "Since ORT 1.9, you are required to explicitly set "
                "the providers parameter when instantiating InferenceSession. For example, "
                f"onnxruntime.InferenceSession(..., providers={available_providers}, ...)"
```

**Comment by ksaur** [comment]
https://github.com/microsoft/onnxruntime/issues/17631  Apparently they are working on a fix

**Comment by ksaur** [comment]
closed with #738 and https://github.com/microsoft/onnxruntime/commit/1c245e6775201fec7a1d269a71deb0f3af6f7bea . Thanks all!
